config/initializers/sidekiq.rb

```diff
@@ -1,6 +1,8 @@
 config = YAML.load(ERB.new(IO.read(Rails.root + 'config' + 'redis.yml')).result)[Rails.env].with_indifferent_access
 
-redis_conn = { url: "redis://#{config[:host]}:#{config[:port]}/", network_timeout: config[:network_timeout] }
+url = Rails.env == 'production' ? "redis://:#{config[:password]}@#{config[:host]}:#{config[:port]}/" : "redis://#{config[:host]}:#{config[:port]}/"
```

Comment:
@afred wanted to draw your attention here due to the change in the url based on the environment.